### DISCUSSION TITLE
Sorting based on dynamic computed value from multiple columns

### DISCUSSION THREAD

**Discussion by updbqn** [TanStack/table#5593] [Q&A]
I have a global filter using @tanstack/match-sorter-utils. I want to sort the rows based on the best matching column.

I have a working solution but it feels a bit hacky. 

I'm adding a hidden column:

```typescript
  {
    id: 'sort',
    accessorFn: () => '',
    sortingFn: globalFuzzySort,
    enableGlobalFilter: false,
  },
```

Where globalFuzzySort sorts each row based on the column with the best match:

```typescript
export const globalFuzzySort = <TData extends RowData>(
  rowA: Row<TData>,
  rowB: Row<TData>,
) => {
  const rankA = Math.max(
    ...Object.values(rowA.columnFiltersMeta).map(
      (meta) => meta?.itemRank?.rank ?? 0,
    ),
  );
  const rankB = Math.max(
    ...Object.values(rowB.columnFiltersMeta).map(
      (meta) => meta?.itemRank?.rank ?? 0,
    ),
  );

  return (
    rankA == rankB ? 0
    : rankA > rankB ? -1
    : 1
  );
};
```

I don't want to use globalFuzzySort for the existing columns directly because I want to fallback on the auto sort when global filter is empty or results in a tie:

```typescript
[ 
       { id: 'sort', desc: false },
       { id: 'col1', desc: false },
       ...
]
```
 
Is there a better approach?

**Comment by updbqn** [comment]
I guess another approach is to fall back on getAutoSortingFn in globalFuzzySort, but what is the best way to get a Column from a row by column id?

```typescript
// I see _getAllCellsByColumnId is private
// perhaps...
row.getAllCells().find((cell) => cell.column.id === columnId)?.column?.getAutoSortingFn()
```

For columns with custom sorting functions I could perhaps specify them in ColumnMeta...

**Comment by btroop** [comment]
dunno if you're still looking for some ideas on this, but you inspired my solution, so maybe i can return the favor

https://github.com/TanStack/table/discussions/5600

**Reply by updbqn** [reply]
Thanks for sharing @btroop. I still think there might be a better way to do it but I'm mostly happy with the solution I arrived at for my current use case.
